pkg/controllers/user/logging/deployer/upgradeimpl.go

```diff
@@ -84,7 +86,7 @@ func (l *loggingService) Upgrade(currentVersion string) (string, error) {
 		return "", fmt.Errorf("get cluster %s failed, %v", l.clusterName, err)
 	}
 	if !v3.ClusterConditionReady.IsTrue(cluster) {
-		return "", fmt.Errorf("upgrade system service %s failed, cluster %v not ready", serviceName, l.clusterName)
+		return "", fmt.Errorf("cluster %v not ready", l.clusterName)
```

Comment:
should this logic be similar to alerting flow? I think that's better, check cluster and catalog conditions only if app actually changed 

or in any case, we should make it consistent for better understanding or comment why it's different

Reply:
We used to check whether old version and latest version in temple for logging and alerting, but because of a bug in chart, we need to redeploy alerting chart without version change, the answer for alerting app is changed, so version comparis on is replaced by app comparison. So the pr keep normal version comparison in logging.

Reply:
got it, thank you